lib/compat/wordpress-6.9/template-activate.php

```diff
@@ -257,8 +257,8 @@ function gutenberg_resolve_block_template( $template_type, $template_hierarchy,
 	// START CORE MODIFICATIONS //
 	//////////////////////////////
 
-	$object            = get_queried_object();
-	$specific_template = $object ? get_page_template_slug( $object ) : null;
+	$object_id         = get_queried_object_id();
+	$specific_template = $object_id && get_post( $object_id ) ? get_page_template_slug( $object_id ) : null;
```

Comment:
```suggestion
	$specific_template = $object_id ? get_page_template_slug( $object_id ) : null;
```

The `get_page_template_slug` also performs a `get_post` check and bails early. See: https://developer.wordpress.org/reference/functions/get_page_template_slug/

Reply:
Oh nice. Perhaps safer to pass an ID instead of a whole queried object though?

Reply:
Yes. ID is probably safer in this case